sklearn/utils/_tags.py

```diff
@@ -393,7 +393,29 @@ def get_tags(estimator) -> Tags:
     tag_provider = _find_tags_provider(estimator)
 
     if tag_provider == "__sklearn_tags__":
-        tags = estimator.__sklearn_tags__()
+        try:
+            tags = estimator.__sklearn_tags__()
+        except AttributeError as exc:
+            if hasattr(estimator, "__sklearn_tags__"):
+                # workaround the regression reported in
+                # https://github.com/scikit-learn/scikit-learn/issues/30479
+                # `__sklearn_tags__` is implemented by calling
+                # `super().__sklearn_tags__()` but there is no `__sklearn_tags__`
+                # method in the base class.
+                warnings.warn(
+                    f"The following error was raised: {str(exc)}. It seems that "
+                    "there are no classes that implement `__sklearn_tags__` "
+                    "in the MRO and/or all classes in the MRO call "
+                    "`super().__sklearn_tags__()`. Make sure to inherit from "
+                    "`BaseEstimator` which implements `__sklearn_tags__`. Note that "
+                    "`BaseEstimator` needs to be on the right side of other Mixins in "
+                    "the inheritance order. The default are now used instead since "
+                    "retrieving tags failed.",
+                    category=DeprecationWarning,
+                )
+                tags = default_tags(estimator)
+            else:
+                raise exc
```

Comment:
I should add a test to cover this line

Reply:
a pragma no cover seems more appropriate here, no ?

Reply:
I added the test instead and find a bug so it was useful :)